### PR TITLE
fix: iavl live migration of orphan nodes

### PR DESCRIPTION
**Pull request by czarcas7ic**
This is cherry picks of work from both John Reynolds and Dev to ensure main stays up to date with the iavl v1.0.0 branch we will be using in v21 of Osmosis.

Previously, we would block chain progress on pruning orphan nodes. For a chain like Osmosis, this took upwards of 2 hours. This change allows chain progress to continue, while pruning orphan nodes synchronously. 

As a drive by change, this also fixes the golang linter in this repo.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -12,11 +12,11 @@ jobs:
...
+        uses: actions/checkout@v4
+      - name: 🐿 Setup Golang
+        uses: actions/setup-go@v4
         with:
-          go-version: '^1.20.0'
+          go-version: 1.21
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3
-        with:
-          version: v1.51.2
```
Comment (czarcas7ic) [inline .github/workflows/lint.yml:22]: This is a drive by change, the linter was not working on any open PR

**Comment by cool-develope** [comment]
Does it include all of #863, #864, #865 changes?

**Comment by czarcas7ic** [comment]
This needs to be reopened with the proper changes just haven't had a moment to do so

**Comment by czarcas7ic** [comment]
Closing in favor of #877 and #876

**Pull request closed without merging**